File: Climbing_Stairs/climbing-stairs.java
// https://leetcode.com/problems/climbing-stairs/


class Solution {
    public int climbStairs(int n) {
        if(n<=1) {
            return 1;
        }
        
        int step_prev1 = 1;
        int step_prev2 = 1;
        int cur = 0;
        
        for(int i=2; i<=n; i++) {
            cur = step_prev1 + step_prev2;
            step_prev1 = step_prev2;
            step_prev2 = cur;
        }
        
        return cur;
    }
}

File: Climbing_Stairs/climbing-stairs.js
// https://leetcode.com/problems/climbing-stairs/

/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function (n) {
  if (n <= 1) {
    return 1;
  }

  let prev_step1 = 1,
    prev_step2 = 1;
  let cur = 0;

  for (let i = 2; i <= n; i++) {
    cur = prev_step1 + prev_step2;
    prev_step1 = prev_step2;
    prev_step2 = cur;
  }
  return cur;
};
